osclib/core.py

```diff
@@ -382,9 +382,22 @@ def repository_path_search(apiurl, project, search_project, search_repository):
 
     return None
 
+def repository_arch_state(apiurl, project, repository, arch):
+    url = makeurl(apiurl, ['build', project, repository, arch, '_repository'],
+                  {'view': 'binaryversions'})
+    from osclib.util import sha1_short
+    return sha1_short(http_GET(url).read())
+
 def repository_state(apiurl, project, repository):
-    return ET.fromstringlist(show_results_meta(
-        apiurl, project, multibuild=True, repository=[repository])).get('state')
+    # Unfortunately, the state hash reflects the published state and not the
+    # binaries published in repository. As such request binary list and hash.
+    url = makeurl(apiurl, ['build', project, repository])
+    root = ETL.parse(http_GET(url)).getroot()
+    combined_state = ''
+    for arch in root.findall('entry'):
+        combined_state += repository_arch_state(apiurl, project, repository, arch.get('name'))
```

Comment:
Is it faster to make multiple calls instead of one?

Reply:
yes, it is